packages/sdk/src/evm/common/any-evm-utils.ts

```diff
@@ -616,6 +691,39 @@ export async function deployInfraWithSigner(
   }
 }
 
+export async function deployPluginsAndMap(
+  signer: Signer,
+  transactions: PrecomputedDeploymentTransaction[],
+  options?: DeployOptions,
+) {
+  let transactionBatches = createTransactionBatches(transactions);
+  if (transactionBatches.length === 0) {
+    return;
+  }
+
+  const deployTxns = await Promise.all(
+    transactionBatches.map((txBatch) => {
+      options?.notifier?.("deploying", "plugin");
+      // Using the deployer contract, send the deploy transactions to common factory with a signer
+      const deployer = new ethers.ContractFactory(
+        DEPLOYER_ABI,
+        DEPLOYER_BYTECODE,
+      )
+        .connect(signer)
+        .deploy(txBatch);
+
+      return deployer;
+    }),
+  );
+
+  await Promise.all(
+    deployTxns.map((tx) => {
+      options?.notifier?.("deployed", "plugin");
+      return tx.deployed();
```

Comment:
why not wait for deplyement confirmation inside the Promise.all above? 

you're also notifying before the actual confirmation here

Reply:
@kumaryash90 ping on this